Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1209,6 +1234,57 @@ instance : IsTotallyReal ℚ where
     rw [Subsingleton.elim v Rat.infinitePlace]
     exact Rat.isReal_infinitePlace
 
+section maximalRealSubfield
+
+open ComplexEmbedding
+
+/--
+The maximal real subfield of `K`. It is totally real, see `isTotallyReal_maximalRealSubfield`, and
+contains all the other totally real subfields of `K`, see `IsTotallyReal.le_maximalRealSubfield`.
+-/
+def maximalRealSubfield : Subfield K where
+  carrier := {x | ∀ φ : K →+* ℂ, star (φ x) = φ x}
+  mul_mem' hx hy _ := by rw [map_mul, star_mul, hx, hy, mul_comm]
+  one_mem' := by simp
+  add_mem' hx hy _ := by rw [map_add, star_add, hx, hy]
+  zero_mem' := by simp
+  neg_mem' := by simp
+  inv_mem' := by simp
+
+instance isTotallyReal_maximalRealSubfield :
+    IsTotallyReal (maximalRealSubfield K) where
+  isReal w := by
+    rw [InfinitePlace.isReal_iff, ComplexEmbedding.isReal_iff]
+    ext x
+    rw [RingHom.star_apply, ← lift_algebraMap_apply K w.embedding]
+    exact x.prop _
+
+variable {K}
+
+theorem IsTotallyReal.le_maximalRealSubfield (E : Subfield K) [h : IsTotallyReal E] :
+    E ≤ maximalRealSubfield K := by
+  intro x hx φ
+  rw [show φ x = (φ.comp E.subtype) ⟨x, hx⟩ by rfl, RCLike.star_def, ← conjugate_coe_eq]
+  refine RingHom.congr_fun ?_ _
+  exact ComplexEmbedding.isReal_iff.mp  <| isReal_mk_iff.mp <| isReal _
+
+theorem isTotallyReal_iff_le_maximalRealSubfield {E : Subfield K} :
+    IsTotallyReal E ↔ E ≤ maximalRealSubfield K :=
+  ⟨fun h ↦ h.le_maximalRealSubfield, fun h ↦ IsTotallyReal.ofRingEquiv
+    (RingEquiv.ofBijective _ (Subfield.inclusion h).rangeRestrictField_bijective).symm⟩
+
+-- Note sure if this should be an instance or a theorem
```

Comment:
Why so? We've got [Subfield.instMin](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Field/Subfield/Basic.html#Subfield.instMin) as an instance, so see no reason why it should not be an instance.

Same comment for the one below.